### PR TITLE
Update Terraform guide for GCP to work with current version of provider

### PR DESCRIPTION
**Pull request by tcaddy**
# Description

Update Terraform guide for GCP to work with current version of provider.

The docs are out of date and do not work with the current TF provider for Port.

I'm not sure exactly how correct I got everything, as I'm brand new to Port.  But at least we can TF plan/apply without error.

## Updated docs pages
- `docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md`

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 1]
```diff
@@ -128,53 +127,48 @@ resource "port_entity" "gcp_folder_entity" {
...
+        title = "Number"
+      }
+      "createTime" = {
+        title  = "Create Time"
+        format = "date-time"
+      }
+    }
+    object_items = {
+      labels = {}
+    }
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:163]: same

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 2]
```diff
@@ -73,53 +73,52 @@ resource "port_blueprint" "gcp_org_blueprint" {
...
-    title      = "Link"
-  }
-  properties {
-    identifier = "createTime"
-    type       = "string"
-    format     = "date-time"
-    title      = "Create Time"
+  properties = {
+    string_items = {
+      link = {
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:78]:
```suggestion
    string_props = {
      link = {
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 3]
```diff
@@ -73,53 +73,52 @@ resource "port_blueprint" "gcp_org_blueprint" {
...
-    format     = "url"
-    title      = "Link"
-  }
-  properties {
-    identifier = "createTime"
-    type       = "string"
-    format     = "date-time"
-    title      = "Create Time"
+  properties = {
+    string_items = {
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:107]:
```suggestion
    string_props = {
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 4]
```diff
@@ -128,53 +127,48 @@ resource "port_entity" "gcp_folder_entity" {
...
-    identifier = "labels"
-    type       = "object"
-    title      = "Labels"
-  }
-  relations {
-    identifier = port_blueprint.gcp_org_blueprint.identifier
-    target = port_blueprint.gcp_org_blueprint.identifier
+  properties = {
+    string_items = {
+      link = {
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:149]:
```suggestion
  properties = {
    string_props = {
      link = {
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 5]
```diff
@@ -128,53 +127,48 @@ resource "port_entity" "gcp_folder_entity" {
...
+      }
+      number = {
+        title = "Number"
+      }
+      "createTime" = {
+        title  = "Create Time"
+        format = "date-time"
+      }
+    }
+    object_items = {
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:161]:
```suggestion
    }
    object_props = {
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 6]
```diff
@@ -183,119 +177,77 @@ resource "port_entity" "gcp_project_entity" {
...
+      "createTime" = each.value.create_time
     }
-  }
-  dynamic "relations" {
-    for_each = each.value.parent.type == "folder" ? [1] : []
-    content {
-      name = port_blueprint.gcp_folder_blueprint.identifier
-      identifier = each.value.parent.id
+    object_items = {
+      labels = each.value.labels
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:187]:
```suggestion
    object_props = {
      labels = each.value.labels
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 7]
```diff
@@ -183,119 +177,77 @@ resource "port_entity" "gcp_project_entity" {
...
+    }
+    object_items = {
+      labels = {}
+      additionalAttributes = {
+        title = "Additional Attributes"
+      }
+      networkTags = {
+        title = "NetworkTags"
+      }
+    }
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:221]:
```suggestion
  properties = {
    string_props = {
      name = {
        title = "name"
      }
      assetType = {
        title = "Asset Type"
      }
      description = {
        title = "Description"
      }
      location = {
        title = "Location"
      }
    }
    object_props = {
      labels = {}
      additionalAttributes = {
        title = "Additional Attributes"
      }
      networkTags = {
        title = "NetworkTags"
      }
    }
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 8]
```diff
@@ -183,119 +177,77 @@ resource "port_entity" "gcp_project_entity" {
...
-    items = each.value.network_tags
+  properties = {
+    string_props = {
+      name        = each.value.name
+      assetType   = each.value.asset_type
+      description = each.value.description
+      location    = each.value.location
+    }
+    object_items = {
+      labels               = each.value.labels
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:242]:
```suggestion
    }
    object_props = {
      labels               = each.value.labels
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 9]
```diff
@@ -406,31 +358,29 @@ resource "port_blueprint" "gcp_org_blueprint" {
...
-    format     = "url"
-    title      = "Link"
-  }
-  properties {
-    identifier = "createTime"
-    type       = "string"
-    format     = "date-time"
-    title      = "Create Time"
+  properties = {
+    string_items = {
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:362]:
```suggestion
  properties = {
    string_props = {
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 10]
```diff
@@ -444,21 +394,22 @@ resource "port_blueprint" "gcp_folder_blueprint" {
...
+      link = {
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:399]:
```suggestion
  properties = {
    string_props = {
      link = {
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 11]
```diff
@@ -491,35 +441,31 @@ resource "port_blueprint" "gcp_project_blueprint" {
...
+        title = "Number"
+      }
+      "createTime" = {
+        title  = "Create Time"
+        format = "date-time"
+      }
+    }
+    object_items = {
+      labels = {}
+    }
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:460]:
```suggestion
  properties = {
    string_props = {
      link = {
        title  = "Link"
        format = "url"
      }
      number = {
        title = "Number"
      }
      "createTime" = {
        title  = "Create Time"
        format = "date-time"
      }
    }
    object_props = {
      labels = {}
    }
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 12]
```diff
@@ -528,36 +474,17 @@ resource "port_entity" "gcp_project_entity" {
...
+      "createTime" = each.value.create_time
     }
-  }
-  dynamic "relations" {
-    for_each = each.value.parent.type == "folder" ? [1] : []
-    content {
-      name = port_blueprint.gcp_folder_blueprint.identifier
-      identifier = each.value.parent.id
+    object_items = {
+      labels = each.value.labels
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:484]:
```suggestion
    object_props = {
      labels = each.value.labels
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 13]
```diff
@@ -571,44 +498,35 @@ resource "port_blueprint" "gcp_asset_blueprint" {
...
+      location = {
+        title = "Location"
+      }
+    }
+    object_items = {
+      labels = {}
+      additionalAttributes = {
+        title = "Additional Attributes"
+      }
+      networkTags = {
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:521]:
```suggestion
    string_props = {
      name = {
        title = "name"
      }
      assetType = {
        title = "Asset Type"
      }
      description = {
        title = "Description"
      }
      location = {
        title = "Location"
      }
    }
    object_props = {
      labels = {}
      additionalAttributes = {
        title = "Additional Attributes"
      }
      networkTags = {
```

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md [hunk 14]
```diff
@@ -620,33 +538,23 @@ resource "port_entity" "gcp_asset_entity" {
...
+    string_props = {
+      name        = each.value.name
+      assetType   = each.value.asset_type
+      description = each.value.description
+      location    = each.value.location
+    }
+    object_items = {
+      labels               = each.value.labels
+      additionalAttributes = each.value.additional_attributes
+      networkTags          = each.value.network_tags
```
Comment (Tankilevitch) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/generic-assets.md:551]:
```suggestion
    string_props = {
      name        = each.value.name
      assetType   = each.value.asset_type
      description = each.value.description
      location    = each.value.location
    }
    object_props = {
      labels               = each.value.labels
      additionalAttributes = each.value.additional_attributes
      networkTags          = each.value.network_tags
```

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Thx for contributing @tcaddy, 
Minor thing I found while while I was looking at the terraform docs is that the properties keys should be [x_props](https://registry.terraform.io/providers/port-labs/port-labs/latest/docs/resources/port_blueprint#nestedatt--properties)  ( e.g instead of string_items => string_props ).

Note that the `x_items` is when specifying [array_props](https://registry.terraform.io/providers/port-labs/port-labs/latest/docs/resources/port_blueprint#nested-schema-for-propertiesarray_props)

**Comment by hadar-co** [comment]
@tcaddy Hi:)
Did you get a chance to look at @Tankilevitch 's comments?

**Pull request closed without merging**